choose-index.md

```diff
@@ -195,197 +195,406 @@ mysql> EXPLAIN SELECT /*+ use_index_merge(t1, idx) */ * FROM t1 WHERE JSON_OVERL
+        |   ├─IndexRangeScan_7(Build)     | 10.00   | cop[tikv] | table:t4, index:mvi2(cast(json_extract(`j`, _utf8'$.b') as unsigned array)) | range:[2,2], keep order:false, stats:pseudo                                                                                                      |
+        |   └─TableRowIDScan_8(Probe)     | 0.01    | cop[tikv] | table:t4                                                                    | keep order:false, stats:pseudo                                                                                                                   |
+        +---------------------------------+---------+-----------+-----------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------------------------+
+        
+        > EXPLAIN SELECT /*+ use_index_merge(t4, mvi1, mvi2) */ * FROM t4 WHERE json_contains(j->'$.a', '[1]') OR json_contains(j->'$.b', '[2]');
+        +-------------------------------+---------+-----------+-----------------------------------------------------------------------------+---------------------------------------------+
+        | id                            | estRows | task      | access object                                                               | operator info                               |
+        +-------------------------------+---------+-----------+-----------------------------------------------------------------------------+---------------------------------------------+
+        | IndexMerge_8                  | 19.99   | root      |                                                                             | type: union                                 |
+        | ├─IndexRangeScan_5(Build)     | 10.00   | cop[tikv] | table:t4, index:mvi1(cast(json_extract(`j`, _utf8'$.a') as unsigned array)) | range:[1,1], keep order:false, stats:pseudo |
+        | ├─IndexRangeScan_6(Build)     | 10.00   | cop[tikv] | table:t4, index:mvi2(cast(json_extract(`j`, _utf8'$.b') as unsigned array)) | range:[2,2], keep order:false, stats:pseudo |
+        | └─TableRowIDScan_7(Probe)     | 19.99   | cop[tikv] | table:t4                                                                    | keep order:false, stats:pseudo              |
+        +-------------------------------+---------+-----------+-----------------------------------------------------------------------------+---------------------------------------------+
+        ```
+
+    - If both `OR` and `AND` appear in the conditions (nested `OR` and `AND`), the conditions that constitute the IndexMerge must all match the semantic of `OR` or the semantic of `AND`, and they cannot make part of them match the semantic of `OR` while the others match the semantic of `AND`. For example:
```

Comment:
"they cannot make part of them match the semantic of `OR` while the others match the semantic" we can improve this